rust/db_handling/src/interface_signer.rs

```diff
@@ -219,9 +219,51 @@ pub fn backup_prep (database_name: &str, seed_name: &str) -> Result<String, Erro
 
 /// Function to prepare key derivation screen.
 /// Gets seed name, network specs key and suggested derivation
-pub fn derive_prep (database_name: &str, seed_name: &str, network_specs_key: &NetworkSpecsKey, suggest: &str) -> Result<String, ErrorSigner> {
+pub fn derive_prep (database_name: &str, seed_name: &str, network_specs_key: &NetworkSpecsKey, collision: Option<(MultiSigner, AddressDetails)>, suggest: &str) -> Result<String, ErrorSigner> {
```

Comment:
This function returns an arbitrary `String` type while in fact it returns some specially-formatted data. It would be better to have a newtype with helpers and constructors that would reflect this on a typesystem level.